query/src/main/java/org/infinispan/query/impl/massindex/IndexWorker.java

```diff
@@ -46,17 +47,17 @@
 public final class IndexWorker implements Function<EmbeddedCacheManager, Void> {
 
    private final String cacheName;
-   private final IndexedTypeIdentifier indexedType;
+   private final Set<IndexedTypeIdentifier> indexedTypes;
    private final boolean flush;
    private final boolean clean;
    private final boolean skipIndex;
    private final boolean primaryOwner;
    private final Set<Object> keys;
 
-   IndexWorker(String cacheName, IndexedTypeIdentifier indexedType, boolean flush, boolean clean, boolean skipIndex,
+   IndexWorker(String cacheName, Set<IndexedTypeIdentifier> indexedTypes, boolean flush, boolean clean, boolean skipIndex,
```

Comment:
Perfect timing to also sort out the TODO on line 76

Reply:
I have upcoming PRs that will get rid of mostly of that decoration